src/main/java/seedu/address/logic/commands/AssignWeddingCommand.java

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * Adds a wedding to an existing person in the Wedlinker.
+ */
+public class AssignWeddingCommand extends Command {
+
+    public static final String COMMAND_WORD = "assign-wedding";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds one or multiple weddings to the person identified "
+            + "by the index number used in the last person listing.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "w/[WEDDING]... (can specify multiple weddings)\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "w/Craig's Wedding w/Wedding April 2025.";
+
+    private final Index index;
+    private final HashSet<Wedding> weddingsToAdd;
+    private boolean force = false;
+
+    /**
+     * Constructs a {@code AssignWedding} Command to add weddings to a person.
+     * @param index The index of the person in the person list.
+     * @param weddingsToAdd The list of weddings to be added.
+     */
+    public AssignWeddingCommand(Index index, HashSet<Wedding> weddingsToAdd) {
+        this.index = index;
+        this.weddingsToAdd = weddingsToAdd;
+    }
+
+    /**
+     * Constructs a {@code AssignWedding} Command to add weddings to a person with the force flag.
+     * @param index The index of the person in the person list.
+     * @param weddingsToAdd The list of weddings to be added.
+     * @param force Whether the command should force the assignment by creating the Wedding object.
+     */
+    public AssignWeddingCommand(Index index, HashSet<Wedding> weddingsToAdd, boolean force) {
+        this.index = index;
+        this.weddingsToAdd = weddingsToAdd;
+        this.force = force;
+    }
+
+    /**
+     * Generates a command execution success message showing the added weddings and the person.
+     *
+     * @param personToEdit The person to whom the weddings were added.
+     * @return A success message indicating the weddings that were added and the name of the person.
+     */
+    private String generateSuccessMessage(Person personToEdit) {
+        String addedWeddings = weddingsToAdd.stream()
+                .map(wedding -> wedding.toString().replaceAll("[\\[\\]]", ""))
+                .collect(Collectors.joining(", "));
+        return String.format(MESSAGE_ADD_WEDDING_SUCCESS, addedWeddings, personToEdit.getName().toString());
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+
+        for (Wedding wedding : weddingsToAdd) {
+            if (!model.hasWedding(wedding)) {
+                if (this.force) {
+                    CreateWeddingCommand newWedding = new CreateWeddingCommand(wedding);
```

Comment:
Maybe can make this variable 'newWeddingCommand' to make the code clearer

Reply:
Fixed.